spec/controllers/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe OmniauthCallbacksController do
+
+  fixtures :users
+
+  describe '#login_dot_gov' do
+    before do
+      mock_user_auth('12345', 'test@gsa.gov')
+      request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:login_dot_gov]
+      get :login_dot_gov
+    end
+
+    let(:user) { User.find_by(email: 'test@gsa.gov') }
+
+    it { is_expected.to redirect_to(admin_home_page_path) }
+
+    context 'when the new user info is saved' do
+      before do
+        get :login_dot_gov
+      end
+
+      it 'saves the email' do
+        expect(user.email).to(eq 'test@gsa.gov')
+      end
+      it 'saves the uid' do
+        expect(user.uid).to(eq '12345')
+      end
+    end
+
+    context 'when the existing user with no uid info is saved' do
+      before do
+        mock_user_auth('12345', 'user_with_out_uid@fixtures.org')
+        get :login_dot_gov
+      end
+
+      let(:user) { User.find_by(email: 'user_with_out_uid@fixtures.org') }
+
+      it 'saves the email' do
+        expect(user.email).to(eq 'user_with_out_uid@fixtures.org')
+      end
+      it 'saves the uid' do
+        expect(user.uid).not_to be_nil
+      end
+    end
+
+    context 'when the existing user with uid info is saved' do
+
+      before do
+        mock_user_auth('11111', 'user_with_uid@fixtures.org')
```

Comment:
See notes above - this auth object is not being used in the spec. The specs are actually just testing the fixture user, rather than anything that the login might be doing.

I'm not sure we even need these specs, because even if the correct auth object were being used in the spec, they would be testing that the login action had no affect on the user object. For an existing user with valid info, I think it would suffice to verify that the user is being assigned to the `@user` instance variable:
```ruby
 it { is_expected.to assign_to(:user).with(user) }
```